### PR TITLE
Rename rdrnd target feature to rdrand

### PR DESCRIPTION
**Pull request by newpavlov**
Plus minor cleanup.

Related stdsimd [issue](https://github.com/rust-lang-nursery/stdsimd/issues/325).

### CODE REVIEW

File: src/librustc_trans/llvm_util.rs [hunk 1]
```diff
@@ -75,26 +75,25 @@ unsafe fn configure_llvm(sess: &Session) {
                                  llvm_args.as_ptr());
 }
 
-// WARNING: the features must be known to LLVM or the feature
-// detection code will walk past the end of the feature array,
-// leading to crashes.
+// WARNING: the features after aplpying `to_llvm_feature` must be known
```
Comment (nikomatsakis) [inline src/librustc_trans/llvm_util.rs]: Typo: s/aplpying/applying

**Comment by nikomatsakis** [comment]
cc @alexcrichton @BurntSushi

**Comment by BurntSushi** [comment]
This seems right to me, and is what Intel refers to upstream. Do we know why LLVM calls this `rdrnd`?

**Comment by nagisa** [comment]
Looking at the blame, it seems that `rdrnd` was chosen to match what gcc and clang "use" (before 2013 the feature was named `rdrand`). Clang probably copied the name from GCC and it is probable that GCC implemented this feature before the RDRAND name was widespread/confirmed/etc.

**Comment by nagisa** [comment]
[This] is the commit.

[This]:https://github.com/rust-lang/llvm/commit/4aa8bdaa46b455d446c10325533dcbdc65eeecdd

**Comment by BurntSushi** [comment]
Ug, so you're saying other compiler *front ends* use `rdrnd` instead of `rdrand`? To me, that makes this a harder decision. So I think our choices are:

1. Conform with what people expect using other popular compilers. (Accept only `rdrnd`.)
2. Conform with upstream vendors. (Accept on `rdrand`.)
3. Conform to both. (Accept `rdrnd` or `rdrand` as aliases of one another.)

I'm not really a fan of any of these, but I feel myself leaning toward (1).

**Comment by newpavlov** [comment]
@alexcrichton [opinion](https://github.com/rust-lang-nursery/stdsimd/pull/320#issuecomment-364680462) was to follow vendor names.

**Comment by nagisa** [comment]
@newpavlov that was an opinion in context of llvm vs vendor and not gcc/clang vs vendor, though.

I can say for sure that I was super surprised that +rdrand didn’t work and took me a while to figure out that it is in fact +rdrnd, so I’m certainly in favour of this change.

**Comment by newpavlov** [comment]
@nagisa 
I believe `pclmul` was chosen by LLVM over `pclmulqdq` for the same reasons as `rdrnd` over `rdrand`. In other words "lvm vs vendor" looks like is equivalent to "gcc/clang vs vendor", as clang follows gcc and in turn LLVM follows clang.

**Comment by BurntSushi** [comment]
I'm still not convinced that this is the right path to take. Being different from the vendor stinks, but being different from other mainstream compiler front ends *also* stinks. I didn't see any consideration given to supporting both?

**Comment by nagisa** [comment]
@BurntSushi well, it is something relevant to more than just rdrand, and is something we should decide about in context of all features. If we decide to have aliases for compatibility with other compilers, we can simply add the cases for all the features at the same time.

**Pull request merged**